akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/sse/LineParser.scala

```diff
@@ -58,7 +58,10 @@ private final class LineParser(maxLineSize: Int) extends GraphStage[FlowShape[By
               case _ ⇒
                 parseLines(bs, from, at + 1, parsedLines)
             }
-        buffer = parseLines(buffer ++ grab(in)) match {
+
+        // start the search where it ended, prevent iterating over all the buffer again
+        val currentBufferStart = if (buffer.isEmpty) 0 else buffer.length - 1
```

Comment:
My suggestion would be `val currentBufferStart = math.max(0, buffer.length - 1)`

Reply:
Updated with your suggestion!